aten/src/ATen/core/dispatch/OperatorOptions.h

```diff
@@ -4,6 +4,30 @@
 
 namespace c10 {
 
+// AliasAnalysisKind is a hint for the TorchScript JIT. It allows the JIT to
+// reason about the return values and arguments passes to registered custom
+// operations.
+//
+// If aliasAnalysisKind is not specified then default is CONSERVATIVE.
+//
+// CONSERVATIVE:TorchScript assumes there are side effects in every input
+// arguments.
+//   "foo(Tensor x, Tensor y, Tensor z) -> Tensor"
+//  there are side effects in x , y and z and return value can alias any
+//  input argument.
+//
+// FROM_SCHEMA: TorchScript uses the SCHEMA to identify which input
+// argument is mutated.
+//     "bar(Tensor(a) x, Tensor y, Tensor z) -> Tensor(a)"
+// x is mutated but not y and z.
```

Comment:
Saw this in src/ATen/native
```
Tensor(a) - a is a set of Tensors that may alias to the same data.
Tensor(a!) - a members of a may be written to thus mutating the underlying data.
```
can x be mutated?

Reply:
Thanks, this is useful. I will add it in this file and clean up some more.